src/components/Pagination/Pagination.js

```diff
@@ -1,23 +1,42 @@
-import ReactPaginate from 'react-paginate';
-import PropTypes from 'prop-types';
+import { useMemo } from "react";
+import ReactPaginate from "react-paginate";
+import PropTypes from "prop-types";
 
-import './Pagination.scss';
+import { Icon } from "../../assets/icons/icons";
+import { getPageNumbers } from "../../services/functions";
+
+import "./Pagination.scss";
+
+export function Pagination({
+  onPageChange,
+  currentPage,
+  productsPerPage,
+  totalProductsCount,
```

Comment:
totalProductsCount => totalPagesCount
productsPerPage => itemsPerPage

try to use Pagination as ui component and not to stick to products as a business value of the items